coherence/coherence_test.go

```diff
@@ -0,0 +1,181 @@
+	utils.AssertEqual(t, err, nil)
+	utils.AssertEqual(t, true, len(val) == 0)
+
+	utils.AssertEqual(t, testStore.Close(), nil)
+}
+
+func Test_Coherence_Get_Missing(t *testing.T) {
+	var val []byte
+
+	testStore, err := newTestStore(t)
+	utils.AssertEqual(t, err, nil)
+
+	val, err = testStore.Get(missingKey)
+	utils.AssertEqual(t, err, nil)
+	utils.AssertEqual(t, true, len(val) == 0)
+
+	utils.AssertEqual(t, testStore.Close(), nil)
+}
+
+func Test_Coherence_Reset(t *testing.T) {
+	var val []byte
+
+	testStore, err := newTestStore(t)
+	utils.AssertEqual(t, err, nil)
+
+	err = testStore.Set(key1, value1, 0)
+	utils.AssertEqual(t, err, nil)
+
+	err = testStore.Set(key2, value2, 0)
+	utils.AssertEqual(t, err, nil)
+
+	// check the keys exist
+	val, err = testStore.Get(key1)
+	utils.AssertEqual(t, err, nil)
+	utils.AssertEqual(t, value1, val)
+
+	val, err = testStore.Get(key2)
+	utils.AssertEqual(t, err, nil)
+	utils.AssertEqual(t, value2, val)
+
+	// reset the store, this should remove both entries
+	err = testStore.Reset()
+
+	// check the keys have expired
+	val, err = testStore.Get(key1)
+	utils.AssertEqual(t, err, nil)
+	utils.AssertEqual(t, true, len(val) == 0)
+
+	val, err = testStore.Get(key2)
+	utils.AssertEqual(t, err, nil)
+	utils.AssertEqual(t, true, len(val) == 0)
+
+	utils.AssertEqual(t, testStore.Close(), nil)
+}
+
+func Test_Coherence_Set_And_Delete(t *testing.T) {
+	var val []byte
+
+	testStore, err := newTestStore(t)
+	utils.AssertEqual(t, err, nil)
+
+	err = testStore.Set(key1, value1, 0)
+	utils.AssertEqual(t, err, nil)
+
+	err = testStore.Delete(key1)
+	utils.AssertEqual(t, err, nil)
+
+	// ensure the key has gone
+	val, err = testStore.Get(key1)
+	utils.AssertEqual(t, err, nil)
+	utils.AssertEqual(t, true, len(val) == 0)
+
+	utils.AssertEqual(t, testStore.Close(), nil)
+}
+
+// TestCoherenceWithScope ensures we can create multiple session stores with multiple scopes.
+func Test_Coherence_With_Scope(t *testing.T) {
+	var val []byte
+
+	// create two session stores with different scopes
+	testStore1, err := newTestStore(t, Config{SessionScope: "scope1"})
+	utils.AssertEqual(t, err, nil)
+
+	testStore2, err := newTestStore(t, Config{SessionScope: "scope2"})
+	utils.AssertEqual(t, err, nil)
+
+	// ensure we can put the same key with different values in each scope
+	err = testStore1.Set(key1, value1, 0)
+	utils.AssertEqual(t, err, nil)
+
+	err = testStore2.Set(key1, value2, 0)
+	utils.AssertEqual(t, err, nil)
+
+	// ensure the value of "key1" is different for each store
+	val, err = testStore1.Get(key1)
+	utils.AssertEqual(t, err, nil)
+	utils.AssertEqual(t, value1, val)
+
+	val, err = testStore2.Get(key1)
+	utils.AssertEqual(t, err, nil)
+	utils.AssertEqual(t, value2, val)
+
+	utils.AssertEqual(t, testStore1.Close(), nil)
+	utils.AssertEqual(t, testStore2.Close(), nil)
+}
+
+// newTestStore returns a new Coherence Store and ensures it is reset.
+func newTestStore(t *testing.T, config ...Config) (*Storage, error) {
```

Comment:
Instead of this, we should add a "Reset" boolean to the config. To allow resetting during creation. It should default to False.